pkg/base/variadic_test.go

```diff
@@ -0,0 +1,339 @@
+		t.Run(test.name, func(t *testing.T) {
+			req := require.New(t)
+
+			result, err := buildStackFromYaml(test.yamlPath, test.yaml)
+			req.NoError(err)
+
+			req.Equal(test.wantStack, result)
+		})
+	}
+}
+
+func Test_buildYamlFromStack(t *testing.T) {
+	tests := []struct {
+		name     string
+		stack    yamlStack
+		wantyaml map[string]interface{}
+	}{
+		{
+			name: "build stack",
+			stack: yamlStack{
+				{
+					NodeName: "",
+					Type:     "map",
+					Index:    0,
+					Data: map[string]interface{}{
+						"spec": map[string]interface{}{
+							"template": map[string]interface{}{
+								"spec": map[string]interface{}{
+									"containers": []interface{}{
+										map[string]interface{}{
+											"name": "hello",
+										},
+									},
+								},
+							},
+						},
+					},
+					Array: nil,
+				},
+				{
+					NodeName: "spec",
+					Type:     "map",
+					Index:    0,
+					Data: map[string]interface{}{
+						"template": map[string]interface{}{
+							"spec": map[string]interface{}{
+								"containers": []interface{}{
+									map[string]interface{}{
+										"name": "hello",
+									},
+								},
+							},
+						},
+					},
+					Array: nil,
+				},
+				{
+					NodeName: "template",
+					Type:     "map",
+					Index:    0,
+					Data: map[string]interface{}{
+						"spec": map[string]interface{}{
+							"containers": []interface{}{
+								map[string]interface{}{
+									"name": "hello",
+								},
+							},
+						},
+					},
+					Array: nil,
+				},
+				{
+					NodeName: "spec",
+					Type:     "map",
+					Index:    0,
+					Data: map[string]interface{}{
+						"containers": []interface{}{
+							map[string]interface{}{
+								"name": "hello",
+							},
+						},
+					},
+					Array: nil,
+				},
+				{
+					NodeName: "containers",
+					Type:     "array",
+					Index:    0,
+					Data: map[string]interface{}{
+						"name": "hello",
+					},
+					Array: []interface{}{
+						map[string]interface{}{
+							"name": "hello",
+						},
+					},
+				},
+			},
+			wantyaml: map[string]interface{}{
+				"spec": map[string]interface{}{
+					"template": map[string]interface{}{
+						"spec": map[string]interface{}{
+							"containers": []interface{}{
+								map[string]interface{}{
+									"name": "hello",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			req := require.New(t)
+
+			result := buildYamlFromStack(test.stack)
+
+			req.Equal(test.wantyaml, result)
+		})
+	}
+}
+
+func Test_parseVariadicTarget(t *testing.T) {
+	tests := []struct {
+		name             string
+		configOptionName string
+		valueName        string
+		target           string
+		want             interface{}
+	}{
+		{
+			configOptionName: "secret",
+			valueName:        "secret-123",
+			target:           "repl{{ ConfigOption repl[[ .secret ]] }}",
+			want:             "repl{{ ConfigOption \"secret-123\" }}",
```

Comment:
template shouldn't return quoted values.  if quotes are wanted, they should be added outside of the template by person writing yaml